keps/sig-autoscaling/1610-container-resource-autoscaling/README.md

```diff
@@ -281,19 +439,614 @@ unit tests will be added:
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+No.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+The operator can query HPAs with `hpa.spec.metrics.containerResource` field set.
+
+###### How can someone using this feature know that it is working for their instance?
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+- [x] Events
+  - `SuccessfulRescale` event with `memory/cpu/etc resource utilization (percentage of request) above/below target`
+    - Note that we cannot know if this reason is due to the `Resource` metric or `ContainerResource` in the current implementation. We'll change this reason for `ContainerResource` to `memory/cpu/etc container resource utilization (percentage of request) above/below target` so that we can distinguish.
+- [x] API .status
+  - When something wrong with the container metrics, `ScalingActive` condition will be false with `FailedGetContainerResourceMetric` reason.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+<!--
+This is your opportunity to define what "normal" quality of service looks like
+for a feature.
+
+It's impossible to provide comprehensive guidance, but at the very
+high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99.9% of /health requests per day finish with 200 code
+
+These goals will help you determine what you need to measure (SLIs) in the next
+question.
+-->
+
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+<!--
+Pick one more of these and delete the rest.
+- [ ] Metrics
+  - Metric name:
+  - [Optional] Aggregation method:
+  - Components exposing the metric:
+- [ ] Other (treat as last resort)
+  - Details:
+-->
+
+N/A
```

Comment:
Do we not have existing SLIs/SLOs for HPA? I would think the same ones would apply. Perhaps something about how quickly we scale when the policy indicates it? Is there really nothing?

Reply:
As far as I know, there is no metrics in the HPA controller. 
But, I may overlook something, please double-check 🙏 @pbetkier @gjtempleton @mwielgus

Reply:
Hey John, we haven't added metrics/SLAs since the graduation of HPA v2 to GA unfortunately, linking back to previous discussion during graduation: https://github.com/kubernetes/enhancements/pull/2703/files#r631955106

Reply:
(Very much agreed that this is something we should have going forward.)

Reply:
Ok...well I didn't block that KEP. For this one, I am going to ask that we fix this debt and add a few metrics here. This will also allow you to resolve @jeremyrickard's point above that metrics informing rollback should not be N/A for beta.

Can you please add a note here and in the response to that question indicating that you will add some metrics as part of beta (and ideally what those metrics are)? I don't think we should keep digging deeper.

Reply:
OK, I added some metrics here. 
Those are just kinda proposed ones from me, but we can discuss and decide the details later in https://github.com/kubernetes/kubernetes/issues/115639